include/oneapi/mkl/types.hpp

```diff
@@ -102,6 +105,26 @@ enum class order : char {
     E = 1,
 };
 
+// Utility function to verify that a given set of types is supported by the
+// device compiler combintiation
+template<typename verify_type, typename T, typename... Ts>
+bool verify_support(cl::sycl::queue q, cl::sycl::aspect aspect){
```

Comment:
oneapi/mkl/types.hpp is a public header, as this function is more related to implementation detail, could you please move it to src/include?

Reply:
Sure, I added a new file `runtime_support_helper.hpp` to the `src/include` folder. I also had to modify the CMakeLists.txt files for the backends so that the `src/include` folder is also among the include directories. I added this change to all the BLAS backends. Should we add it to the other domains as well? Let me know what you think about this change and in case you don't want to have a new file for only this function where it would fit well.

Reply:
Thanks @sbalint98! So far only BLAS has half type API, so there is no need to add this header to other domains. New file looks good to me, I don't think it fits to any files that we have in this folder.

Could you please run clang-format for new file and `src/blas/backends/cublas/cublas_level3.cpp`?

Reply:
Sure. I keep forgetting that.

Reply:
Thanks!